helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/AssignmentMetadataStore.java

```diff
@@ -146,6 +146,17 @@ public boolean persistBestPossibleAssignment(
     return true;
   }
 
+  protected synchronized void resetCache() {
```

Comment:
Nit: I think `clearMetadataStore()` might be a better name for this method because caching isn't really an explicit feature? Or resetMetadataStore()?

Reply:
resetMetadataStore it is. Good suggestion.